bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/DialogProcessor.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.voice.internal;
+
+import java.util.HashSet;
+import java.util.Locale;
+
+import org.eclipse.smarthome.core.audio.AudioException;
+import org.eclipse.smarthome.core.audio.AudioSink;
+import org.eclipse.smarthome.core.audio.AudioSource;
+import org.eclipse.smarthome.core.audio.AudioStream;
+import org.eclipse.smarthome.core.audio.UnsupportedAudioFormatException;
+import org.eclipse.smarthome.core.voice.KSErrorEvent;
+import org.eclipse.smarthome.core.voice.KSEvent;
+import org.eclipse.smarthome.core.voice.KSException;
+import org.eclipse.smarthome.core.voice.KSListener;
+import org.eclipse.smarthome.core.voice.KSService;
+import org.eclipse.smarthome.core.voice.KSpottedEvent;
+import org.eclipse.smarthome.core.voice.RecognitionStopEvent;
+import org.eclipse.smarthome.core.voice.STTEvent;
+import org.eclipse.smarthome.core.voice.STTException;
+import org.eclipse.smarthome.core.voice.STTListener;
+import org.eclipse.smarthome.core.voice.STTService;
+import org.eclipse.smarthome.core.voice.STTServiceHandle;
+import org.eclipse.smarthome.core.voice.SpeechRecognitionErrorEvent;
+import org.eclipse.smarthome.core.voice.SpeechRecognitionEvent;
+import org.eclipse.smarthome.core.voice.TTSException;
+import org.eclipse.smarthome.core.voice.TTSService;
+import org.eclipse.smarthome.core.voice.Voice;
+import org.eclipse.smarthome.core.voice.text.HumanLanguageInterpreter;
+import org.eclipse.smarthome.core.voice.text.InterpretationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An instance of this class can handle a complete dialog with the user. It orchestrates the keyword spotting, the stt
+ * and tts services together with the human language interpreter.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ */
+public class DialogProcessor implements KSListener, STTListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(DialogProcessor.class);
```

Comment:
We could rely on a non-static logger.
I will not repeat this for every one for now.